src/main/java/core/basesyntax/JavaDateTimeApi.java

```diff
@@ -1,125 +1,158 @@
 package core.basesyntax;
 
+import java.time.DateTimeException;
+import java.time.Instant;
 import java.time.LocalDate;
 import java.time.LocalDateTime;
 import java.time.LocalTime;
 import java.time.OffsetDateTime;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.util.Locale;
 import java.util.Optional;
 
 public class JavaDateTimeApi {
+    private static final String TIME_ZONE = "+02:00";
+    private static final DateTimeFormatter DATE_TIME_FORMATTER
+            = DateTimeFormatter.ofPattern("d MMM yyyy", Locale.UK);
+
     /**
      * Return the current date as a String depending on a query.
-     *
+     * <p>
      * The query can be passed for the whole date or for it's part:
-     *      - FULL - current date as a whole: year, month, day of month
-     *        formatted as `YYYY-MM-DD` (a default return value);
-     *      - YEAR - current year;
-     *      - MONTH - name of the current month;
-     *      - DAY - current day of month;
+     * - FULL - current date as a whole: year, month, day of month
+     * formatted as `YYYY-MM-DD` (a default return value);
+     * - YEAR - current year;
+     * - MONTH - name of the current month;
+     * - DAY - current day of month;
      * In any other case throw DateTimeException.
      **/
     public String todayDate(DateTimePart datePart) {
-        return "Today";
+        switch (datePart) {
+            case FULL:
+                return String.valueOf(LocalDate.now());
+            case YEAR:
+                return String.valueOf(LocalDate.now().getYear());
+            case MONTH:
+                return String.valueOf(LocalDate.now().getMonth());
+            case DAY:
+                return String.valueOf(LocalDate.now().getDayOfMonth());
+            default:
+                throw new DateTimeException("Wrong date!");
+        }
     }
 
     /**
      * Given an Array of 3 elements, where
-     *         - 1-st element is a `year`;
-     *         - 2-nd element is s `month`;
-     *         - 3-rd element is a `day of month`;
-     *
+     * - 1-st element is a `year`;
+     * - 2-nd element is s `month`;
+     * - 3-rd element is a `day of month`;
+     * <p>
      * Return Optional of a date built from these elements.
      */
     public Optional<LocalDate> getDate(Integer[] dateParams) {
-        return Optional.empty();
+        return dateParams.length == 0 || dateParams[1] > 12 ? Optional.empty()
+                : Optional.of(LocalDate.of(dateParams[0], dateParams[1], dateParams[2]));
```

Comment:
Let's catch here two exceptions IndexOutOfBoundsException and DateTimeException https://mate-academy.github.io/jv-program-common-mistakes/java-core/java-eight-part-two/date-time-api#we-can-catch-two-particular-exceptions-in-the-same-catch-block. You won't check each specific case everytime

Reply:
Changed.